### PR TITLE
Mccalluc/test tsv to mrmatrix

### PR DESCRIPTION
**Pull request by mccalluc**
No actual tests, yet, but if this is misguided, better to know sooner than later. I'll continue to work on this branch, and mark it "read for review" in github when done.

### CODE REVIEW

**Comment by mccalluc** [comment]
Assuming this is the right direction, I'd like to do more, but this much is coherent.

**Comment by mccalluc** [comment]
hmm: passing locally, but on Travis:
```
Traceback (most recent call last):
  File "/home/travis/build/higlass/clodius/test/tsv_to_mrmatrix_test.py", line 15, in test_something
    hdf5 = h5py.File(TemporaryFile(), 'w')
  File "/home/travis/miniconda/lib/python3.6/site-packages/h5py/_hl/files.py", line 267, in __init__
    name = name.encode(sys.getfilesystemencoding())
AttributeError: '_io.BufferedRandom' object has no attribute 'encode'
```
I'll look into this... some difference in environment? Might try using a normal file, and see if that makes a difference.

**Comment by mccalluc** [comment]
I'm seeing that aggregation is based on summing the cells of the previous layer, and I think that's just not what I was expecting. The HiC data is counts, so that makes sense, but for my image data, where I think I'd want to preserve the idea that each pixel is just scaled between 0 and 1, I'd probably want either a mean, or possibly max?

Does it feel like the aggregation function should be passed in as a parameter? I could be thinking about it wrong.

File: scripts/tsv_to_mrmatrix.py [hunk 1]
```diff
@@ -42,60 +42,37 @@ def coarsen(f):
...
+def parse(input_handle, output_hdf5, top_n=None):
+    input_handle
+    first_line = next(input_handle)
+    parts = first_line.strip().split('\t')
+    # TODO: Use the python built-in csv module, instead of parsing by hand?
 
     if top_n is None:
         top_n = len(parts) - 1
+        # TODO: So if it's taller than it is wide, it will be truncated to a square,
+        # unless an explicit top_n is provided? That doesn't seem right.
```
Comment (pkerpedjiev) [inline scripts/tsv_to_mrmatrix.py]: You're right. I wrote this script with square matrices in mind. Extending it to rectangular matrices will need some fixing. Having a TODO is a nice way of kicking that can down the road.

**Pull request merged**